### PR TITLE
Move logic to hide Template selector in Shop page into Product Catalog class and add e2e test

### LINKED ISSUE #46376: Add test to verify Template selector is hidden in the Shop page editor
We intentionally hide the `Template` selector in the _Shop_ page editor as it doesn't have any effect (its template is handled by the Site Editor, see https://github.com/woocommerce/woocommerce-blocks/issues/6278 for more details).

This issue is about adding some automated tests that make sure there are no regressions and the `Template` doesn't show up by mistake.

### PR DESCRIPTION
**Pull request by Aljullu**
### Changes proposed in this Pull Request:

This PR moves the logic to hide the _Template_ selector when editing the _Shop_ page from the `BlockTemplatesController` classic into the `ProductCatalogTemplate` class. Additionally, it adds an e2e test to make sure there are no regressions in the future.

Closes https://github.com/woocommerce/woocommerce/issues/46376.

### How to test the changes in this Pull Request:

This PR doesn't introduce any functionality changes, so test is simply making sure there are no regressions.

1. Go to Pages > Shop.
2. Verify there is no _Template_ selector in the sidebar.
3. Edit any other page.
4. Verify there is the _Template_ selector in the sidebar.

Shop page | Any other page
--- | ---
imatge | imatge

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/templates/shop-page.block_theme.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		);
+		const lines = cliOutput.stdout.split( '\n' );
+		const shopPageId = lines.reverse().find( ( line ) => line !== '' );
+
+		// eslint-disable-next-line playwright/no-conditional-in-test
+		if ( ! shopPageId ) {
+			throw new Error( 'Shop page ID not found' );
+		}
+
+		await admin.editPost( shopPageId );
```
Comment (WunderBart) [inline plugins/woocommerce-blocks/tests/e2e/tests/templates/shop-page.block_theme.spec.ts]: Maybe we could use regexp here for clarity, e.g.:

```suggestion
		const numberMatch = cliOutput.stdout.match( /\d+/ );
		expect( numberMatch ).not.toBeNull();
		
		await admin.editPost( numberMatch[0] );
```

File: plugins/woocommerce-blocks/tests/e2e/tests/templates/shop-page.block_theme.spec.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+		await expect( page.getByText( 'Template' ) ).toHaveCount( 0 );
```
Comment (WunderBart) [inline plugins/woocommerce-blocks/tests/e2e/tests/templates/shop-page.block_theme.spec.ts]: Let's assert the visibility instead (as per the test description):

```suggestion
		await expect( page.getByText( 'Template' ) ).toBeHidden();
```

**Review by WunderBart — APPROVED** [review]
Tested locally, didn't spot any regressions! 👍 

Left 2 non-blocking suggestions. 🚀

**Comment by Aljullu** [comment]
Thanks for the review, @WunderBart! I applied both of your suggestions, so merging. :ship:

**Pull request merged**